docs/build/use-cases/key-generation.mdx

```diff
@@ -0,0 +1,576 @@
+---
+description: Generate an EVM key inside ROFL via appd and use it to sign and send transactions on Base.
+tags: [ROFL, appd, keys, EVM]
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# ROFL Key Generation (EVM / Base)
+
+This chapter shows how to build a tiny TypeScript app that **generates a
+secp256k1 key inside ROFL** via the [appd REST API], derives an **EVM
+address**, and **signs & sends** EIP‑1559 transactions on **Base Sepolia**.
+We also include an optional **contract deployment** step.
+
+[appd REST API]: ../rofl/features/appd.md
+
+## Prerequisites
+
+This guide requires:
+
+- **Node.js 20+** and **Docker** (or Podman).
+- **Oasis CLI** and at least **120 TEST** tokens in your wallet.
+- A Base Sepolia faucet or funds to test sending ETH.
+
+Check [Quickstart Prerequisites] for setup details.
+
+[Quickstart Prerequisites]: ../rofl/quickstart#prerequisites
+
+## Init App
+
+Initialize a new app using the [Oasis CLI]:
+
+```shell
+oasis rofl init rofl-keygen
+cd rofl-keygen
+````
+
+## Create App
+
+Create the app on Testnet (100 TEST deposit):
+
+```shell
+oasis rofl create --network testnet
+```
+
+The CLI prints the **App ID** (e.g., `rofl1...`). We will expose it
+via an HTTP endpoint as well.
+
+## Install minimal deps
+
+We keep dependencies lean:
+
+```shell
+npm init -y
+npm i express ethers zod dotenv
+npm i -D typescript tsx @types/node @types/express hardhat
+npx tsc --init --rootDir src --outDir dist --module NodeNext --target ES2022
```

Comment:
I'd just use `hardhat init` and select typescript. Then install any extra deps. See how we did it for Sapphire: https://github.com/oasisprotocol/sapphire-paratime/blob/main/docs/quickstart.mdx#init-a-new-hardhat-project